frontend/packages/knative-plugin/src/components/traffic-splitting/TrafficModalRevisionsDropdownField.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import { set, cloneDeep } from 'lodash';
+import { useField } from 'formik';
+import { DropdownField } from '@console/shared';
+import { RevisionItems } from '../../utils/traffic-splitting-utils';
+
+type TrafficModalRevisionsDropdownFieldProps = {
+  revisionItems: RevisionItems;
+  name: string;
+  title: string;
+};
+
+const TrafficModalRevisionsDropdownField: React.FC<TrafficModalRevisionsDropdownFieldProps> = ({
+  revisionItems,
+  name,
+  title,
+}) => {
+  const [field] = useField(name);
+  const dropdownItems =
+    !field.value || Object.values(revisionItems).includes(field.value)
+      ? revisionItems
+      : set(cloneDeep(revisionItems), [field.value], field.value);
```

Comment:
nit: we could simply do this and remove lodash imports.
```suggestion
      : {...revisionItems, [field.value] : field.value};
```

Reply:
done